src/EnergyPlus/ConvectionCoefficients.cc

```diff
@@ -2128,32 +2017,37 @@ Real64 CalcZoneSupplyAirTemp(EnergyPlusData &state, int const ZoneNum)
 {
 
     int ZoneNode = state.dataHeatBal->Zone(ZoneNum).SystemZoneNodeNumber;
-    if (ZoneNode > 0) {
-        int thisZoneInletNode = 0;
-        Real64 SumMdotTemp = 0.0;
-        Real64 SumMdot = 0.0;
-        for (int EquipNum = 1;
-             EquipNum <= state.dataZoneEquip->ZoneEquipList(state.dataZoneEquip->ZoneEquipConfig(ZoneNum).EquipListIndex).NumOfEquipTypes;
-             ++EquipNum) {
-            if (state.dataZoneEquip->ZoneEquipList(state.dataZoneEquip->ZoneEquipConfig(ZoneNum).EquipListIndex).EquipData(EquipNum).NumOutlets > 0) {
-                thisZoneInletNode = state.dataZoneEquip->ZoneEquipList(state.dataZoneEquip->ZoneEquipConfig(ZoneNum).EquipListIndex)
-                                        .EquipData(EquipNum)
-                                        .OutletNodeNums(1);
-                if ((thisZoneInletNode > 0) && (state.dataLoopNodes->Node(thisZoneInletNode).MassFlowRate > 0.0)) {
-                    SumMdotTemp += state.dataLoopNodes->Node(thisZoneInletNode).MassFlowRate * state.dataLoopNodes->Node(thisZoneInletNode).Temp;
-                    SumMdot += state.dataLoopNodes->Node(thisZoneInletNode).MassFlowRate;
-                }
-            }
-        }
-        if (SumMdot > 0.0) {
-            return SumMdotTemp / SumMdot; // mass flow weighted inlet temperature
-        } else {
-            if (thisZoneInletNode > 0) {
-                return state.dataLoopNodes->Node(thisZoneInletNode).Temp;
-            } else {
-                return state.dataLoopNodes->Node(ZoneNode).Temp;
-            }
+    if (ZoneNode <= 0)
+        return state.dataLoopNodes->Node(ZoneNode).Temp;
```

Comment:
This can't be right. I see the current code does this, too, but this will violate array bounds. I think this should be one of the zone air temperature variables (not sure which one). This needs a separate PR to fix.

Reply:
Wonder why this never triggered before ... 🤔